### PR TITLE
fix key of connectedJoyCons, productId → device serial id

### PR DESCRIPTION
**Pull request by taisukef**
for same productId two or more devices

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1,21 +1,38 @@
 import { JoyConLeft, JoyConRight, GeneralController } from './joycon.js';
 
 const connectedJoyCons = new Map();
+const devices = [];
+const getDeviceID = (device) => {
```
Comment (tomayac) [inline src/index.js:5]:
```suggestion

const getDeviceID = (device) => {
```

File: src/index.js [hunk 2]
```diff
@@ -1,21 +1,38 @@
...
+  const n = devices.indexOf(device);
+  if (n >= 0) {
+    return n;
+  }
+  devices.push(device);
+  return devices.length - 1;
+};
+const addDevice = async (device) => {
```
Comment (tomayac) [inline src/index.js:13]:
```suggestion

const addDevice = async (device) => {
```

File: src/index.js [hunk 3]
```diff
@@ -1,21 +1,38 @@
...
+  const id = getDeviceID(device);
+  console.log(`HID connected: ${id} ${device.productName}`);
+  connectedJoyCons.set(id, await connectDevice(device));
+};
+const removeDevice = async (device) => {
```
Comment (tomayac) [inline src/index.js:18]:
```suggestion

const removeDevice = async (device) => {
```

File: src/index.js [hunk 4]
```diff
@@ -57,18 +74,18 @@ const connectJoyCon = async () => {
...
 };
 
 const connectDevice = async (device) => {
-  let joyCon;
+  let joyCon = null;
   if (device.productId === 0x2006) {
     joyCon = new JoyConLeft(device);
   } else if (device.productId === 0x2007) {
-    if (device.productName.startsWith('Joy-Con (R)')) {
+    if (device.productName == 'Joy-Con (R)') {
```
Comment (tomayac) [inline src/index.js:88]:
```suggestion
    if (device.productName === 'Joy-Con (R)') {
```

**Review by tomayac — APPROVED** [review]
Just some super tiny nits. After fixing those: LGTM

**Comment by taisukef** [comment]
Thank you for you review!

**Review by tomayac — APPROVED** [review]
LGTM

**Pull request merged**